packages/flutter/test/widgets/form_test.dart

```diff
@@ -578,4 +578,232 @@ void main() {
+                  initialValue: 'foo',
+                  autoValidateMode: AutoValidateMode.onUserInteraction,
+                  builder: (FormFieldState<String> state) {
+                    formFieldState = state;
+                    return Container();
+                  },
+                  validator: errorText,
+                ),
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(builder());
+    // The form field has no error.
+    expect(formFieldState.hasError, isFalse);
+    // No error widget is visible.
+    expect(find.text(errorText('foo')), findsNothing);
+  });
+
+  testWidgets('auto-validate before value changes if autoValidateMode was set to always', (WidgetTester tester) async {
+    FormFieldState<String> formFieldState;
+
+    String errorText(String value) => '$value/error';
+
+    Widget builder() {
+      return MaterialApp(
+        home: MediaQuery(
+          data: const MediaQueryData(devicePixelRatio: 1.0),
+          child: Directionality(
+            textDirection: TextDirection.ltr,
+            child: Center(
+              child: Material(
+                child: FormField<String>(
+                  initialValue: 'foo',
+                  autoValidateMode: AutoValidateMode.always,
+                  builder: (FormFieldState<String> state) {
+                    formFieldState = state;
+                    return Container();
+                  },
+                  validator: errorText,
+                ),
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(builder());
+    expect(formFieldState.hasError, isTrue);
+  });
+
+  testWidgets('Form auto-validate form fields only after one of them changes if autovalidateMode is onUserInteraction', (WidgetTester tester) async {
+    const String initialValue = 'foo';
+    String errorText(String value) => 'error/$value';
+
+    Widget builder() {
+      return MaterialApp(
+        home: Directionality(
+          textDirection: TextDirection.ltr,
+          child: Center(
+            child: Material(
+              child: Form(
+                autoValidateMode: AutoValidateMode.onUserInteraction,
+                child: Column(
+                  children: <Widget>[
+                    TextFormField(
+                      initialValue: initialValue,
+                      validator: errorText,
+                    ),
+                    TextFormField(
+                      initialValue: initialValue,
+                      validator: errorText,
+                    ),
+                    TextFormField(
+                      initialValue: initialValue,
+                      validator: errorText,
+                    )
+                  ],
+                ),
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+
+    // The issue only happens on the second build so we need to rebuild the three.
```

Comment:
This sentence is a bit confusing to people who do not know the context, maybe `Makes sure it won't autovalidate after rebuilds if there is no user interaction`

Reply:
Here I'm just trying to explain why I called pumpWidget twice

Reply:
I understand, other people might not know what is the issue you are referring here. Can you add a sentence or two to explain it?

Reply:
Right

Reply:
> This sentence is a bit confusing to people who do not know the context, maybe `Makes sure it won't autovalidate after rebuilds if there is no user interaction`

I will replace the test desciption with your sentence since its more clear for the users